packages/clerk-js/src/ui/router/Route.tsx

```diff
@@ -25,18 +26,16 @@ type GuardedRouteProps = {
 
 export type RouteProps = React.PropsWithChildren<UnguardedRouteProps | GuardedRouteProps>;
 
-const RouteGuard = ({
-  canActivate,
-  fallbackPath,
-  children,
-}: React.PropsWithChildren<RouteGuardProps>): JSX.Element | null => {
-  const router = useRouter();
+const RouteGuard = ({ canActivate, children }: React.PropsWithChildren<RouteGuardProps>): JSX.Element | null => {
+  const { navigateToFlowStart } = useNavigateToFlowStart();
+  const clerk = useCoreClerk();
+
   React.useEffect(() => {
-    if (!canActivate()) {
-      void router.navigate(fallbackPath);
+    if (!canActivate(clerk)) {
```

Comment:
❓ Is `canActivate` always defined? It looks like it might be undefined in some cases: https://github.com/clerkinc/javascript/pull/1491/files#diff-1d0e1700c66dc695d57f40de333fcaeaba78e7cdd0dbe8fb28015d5a501aad4fR121

Reply:
We only use it if it is defined, yes.